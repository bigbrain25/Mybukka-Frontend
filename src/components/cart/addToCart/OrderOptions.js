import React from 'react';
import { useMediaQuery } from 'react-responsive';
import FooterBigScreen from '../../footer/FooterBigScreen';
import ProductList from '../../product-list/ProductList';
import FooterSmallScreen from '../../footer/FooterSmallScreen';
import ProductDescription from '../../product-description/ProductDescription';

import './OrderOptions.scss';
import { useCartContext } from '../../../context/CartContext';

const OrderOptions = ({ toggleAddToCart, mealToDisplay, storeIntoCart }) => {
  const isBigScreen = useMediaQuery({ minWidth: 960 });
  const { price, title, description } = mealToDisplay;
  const { inProgressCart: { totalCost, originalCost } } = useCartContext();

  return (
    <div className="OrderOptions">
      <div className="OrderOptions-Top">
        <ProductDescription title={title} desc={description} toggleAddToCart={toggleAddToCart} />
        <ProductList products={mealToDisplay} />
        <FooterSmallScreen price={price} />
      </div>
      {isBigScreen && <FooterBigScreen withQtyBtn handleClick={storeIntoCart} totalCost={totalCost + (originalCost || price)} price={price} />}
    </div>
  );
};

export default OrderOptions;
