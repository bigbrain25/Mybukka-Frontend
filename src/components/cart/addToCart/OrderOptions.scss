.OrderOptions {
  position: relative;
  height: 100%;
  width: 100%;
}

.OrderOptions-Top {
  position: relative;
  height: 100%;
  @media screen and (min-width: 1060px) {
    height: calc(100% - 88px);
    overflow-y: auto;
    border-bottom: 1px solid rgba(217, 219, 224, 0.5);
  }
}
