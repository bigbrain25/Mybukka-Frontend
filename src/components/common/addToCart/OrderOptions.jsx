import React from 'react';
import PropTypes from 'prop-types';


import Column from 'Components/grid/Column';
import Cancel from 'Components/icons/Cancel';
import { MealTitle, MealDescription } from 'Components/card/MealCard';

import SpecialInstructions from './SpecialInstructions';
import SubMenus from './SubMenus';
import ActionSection from './ActionSection';

import './orderOptions.scss';

const OrderOptions = ({ title, description, price, slug, quantity, submenus }) => (
  <Column classNames="col-12 col-xs-12 col-sm-12 col-md-12 col-lg-5 order-options-section">
    <div
      data-dismiss="modal"
      className="dismiss-modal-options d-none d-lg-block"
    >
      <Cancel />
    </div>
    <div className="options-details">
      <MealTitle title={title} />
      <MealDescription description={description} />
      <SpecialInstructions />
      {submenus.length > 0 && submenus.map(eachMenu => (
        <SubMenus menus={eachMenu.options} title={eachMenu.title} />
      ))}
    </div>
    <ActionSection price={price} slug={slug} quantity={quantity} />
  </Column>
);

export default OrderOptions;

OrderOptions.defaultProps = {
  description: ''
};

OrderOptions.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  price: PropTypes.number.isRequired,
  slug: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
};
