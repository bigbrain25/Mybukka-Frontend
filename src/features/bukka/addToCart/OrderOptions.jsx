import React from 'react';
import PropTypes from 'prop-types';
import Column from 'Components/grid/Column';

import Cancel from 'Components/icons/Cancel';

import SpecialInstructions from './SpecialInstructions';
import ActionSection from './ActionSection';
import { MealTitle, MealDescription } from '../components/MealCard';

import './orderOptions.scss';

const OrderOptions = ({ title, description }) => (
  <Column classNames="col-12 col-xs-12 col-sm-12 col-md-12 col-lg-5 order-options-section">
    <div
      data-dismiss="modal"
      className="dismiss-modal-options d-none d-lg-block"
    >
      <Cancel />
    </div>
    <div className="options-details">
      <MealTitle title={title} />
      <MealDescription description={description} />
      <SpecialInstructions />
    </div>
    <ActionSection price={13} handleClick={() => {}} />
  </Column>
);

export default OrderOptions;

OrderOptions.defaultProps = {
  description: ''
};

OrderOptions.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string
};
