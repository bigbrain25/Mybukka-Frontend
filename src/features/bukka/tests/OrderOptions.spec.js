import React from 'react';
import OrderOptions from '../addToCart/OrderOptions';

describe('Order Options Component', () => {
  const props = {
    title: 'Amala',
    descriptions: 'best meal youll ever eat'
  };
  const wrapper = mount(<OrderOptions {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
