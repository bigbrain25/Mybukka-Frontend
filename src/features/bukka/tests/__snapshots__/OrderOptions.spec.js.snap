// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Options Component renders properly 1`] = `ReactWrapper {}`;
